engine/api/api_routes.go

```diff
@@ -266,6 +266,10 @@ func (api *API) InitRouter() {
 	r.Handle("/artifact/store", r.GET(api.getArtifactsStoreHandler, Auth(false)))
 	r.Handle("/artifact/{hash}", r.GET(api.downloadArtifactDirectHandler, Auth(false)))
 
+	// Cache
+	r.Handle("/project/{key}/cache/{tag}/push", r.POSTEXECUTE(api.postPushCacheHandler, NeedWorker()))
```

Comment:
why not only one route /project/{key}/cache/{tag} and two methods POST and GET to handle push and pull ?

Reply:
i think we need a route to post with a tempurl